src/vminfo.h

```diff
@@ -0,0 +1,150 @@
+/*
+ * vminfo.h - parse libvirt bulk stats records into C structures
+ * Copyright (C) 2014-2016 Red Hat, Inc.
+ * Written by Francesco Romani <fromani@redhat.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program;
+ * if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#ifndef VMINFO_H
+#define VMINFO_H
+
+#include <stdio.h>
+
+#include <libvirt/libvirt.h>
+
+enum {
+  STATS_NAME_LEN = 128,
+  BLOCK_STATS_NUM = 8,
+  IFACE_STATS_NUM = 8,
+  VCPU_STATS_NUM = 16
+};
+
+typedef struct BlockStats BlockStats;
+struct BlockStats {
+  char *xname;
+  char name[STATS_NAME_LEN];
+
+  unsigned long long rd_reqs;
+  unsigned long long rd_bytes;
+  unsigned long long rd_times;
+  unsigned long long wr_reqs;
+  unsigned long long wr_bytes;
+  unsigned long long wr_times;
+  unsigned long long fl_reqs;
+  unsigned long long fl_times;
+
+  unsigned long long allocation;
+  unsigned long long capacity;
+  unsigned long long physical;
+};
+
+typedef struct BlockInfo BlockInfo;
+struct BlockInfo {
+  size_t nstats;
+  BlockStats *xstats;
+  BlockStats stats[BLOCK_STATS_NUM];
+};
+
+typedef struct IFaceStats IFaceStats;
+struct IFaceStats {
+  char *xname;
+  char name[STATS_NAME_LEN];
+
+  unsigned long long rx_bytes;
```

Comment:
Seems like you are redefining libvirt structure here. Any reason not to use in this case virDomainInterfaceStatsStruct? This comment also applies to other structures like BlockStats.